File: tests/fixtures/capture-helper.cs
// Test-only protocol fixture. It does not install hooks or inspect any application.
using System;
using System.IO;
using System.Web.Script.Serialization;
class Fixture {
  static void Main(string[] args) {
    if(args.Length>0 && args[0]=="--focus"){Console.WriteLine("focused");return;}
    Console.OutputEncoding=new System.Text.UTF8Encoding(false);
    var json=new JavaScriptSerializer();int id=0;bool active=false;
    Action<object> send=value=>{Console.WriteLine(json.Serialize(value));Console.Out.Flush();};
    send(new {type="ready"});string line;
    while((line=Console.ReadLine())!=null){
      if(line=="enable"||line=="disable")send(new {type="status",enabled=line=="enable",error=""});
      else if(line=="capture"){
        if(active){send(new {type="raise"});continue;}
        active=true;id++;send(new {type="begin",id=id});
        string mode=Path.Combine(AppDomain.CurrentDomain.BaseDirectory,"mode.txt");
        bool empty=File.Exists(mode)&&File.ReadAllText(mode)=="empty";
        send(new {type="result",id=id,text=empty?"":"Prova Localino \ud83c\udf31\nSeconda riga \u00e8 \u6f22\u5b57",reason=empty?"empty":"ok",ms=5});
      }
      else if(line.StartsWith("visible:"))send(new {type="timing",id=id,ms=42});
      else if(line=="finish"||line=="cancel")active=false;
      else if(line=="quit")return;
    }
  }
}

File: tests/fixtures/capture-helper.mjs
#!/usr/bin/env node
// Test-only protocol fixture; never installs a hook or reads another application.
import {createInterface} from 'node:readline'
import {readFileSync} from 'node:fs'
import {dirname,join} from 'node:path'
import {fileURLToPath} from 'node:url'
if(process.argv[2]==='--focus'){process.stdout.write('focused\n');process.exit(0)}
const send=value=>process.stdout.write(JSON.stringify(value)+'\n')
let active=false,id=0;send({type:'ready'})
createInterface({input:process.stdin}).on('line',line=>{
  if(line==='enable'||line==='disable')send({type:'status',enabled:line==='enable',error:''})
  else if(line==='capture'){
    if(active){send({type:'raise'});return}active=true;id++;send({type:'begin',id})
    let empty=false;try{empty=readFileSync(join(dirname(fileURLToPath(import.meta.url)),'mode.txt'),'utf8')==='empty'}catch{}
    send({type:'result',id,text:empty?'':'Prova Localino 🌱\nSeconda riga è 漢字',reason:empty?'empty':'ok',ms:5})
  }else if(line.startsWith('visible:'))send({type:'timing',id,ms:42})
  else if(line==='cancel'||line==='finish')active=false
}).on('close',()=>process.exit(0))
